### DISCUSSION TITLE
MarkerType arrow or arrowclosed not showing.

### DISCUSSION THREAD

**Discussion by hiimnhan** [xyflow/xyflow#3772] [Q&A]
Hi, I made a custom edge with `markerEnd` set to `MarkerType.ArrowClosed` but it's not showing up. Do I miss something?
Here is the `CustomEdge` component
```
import { EdgeProps, MarkerType } from "reactflow";
import CustomEdge from "../CustomEdge";
import { CSSProperties } from "react";

type OneWayArrowProps = EdgeProps & {
  stroke?: CSSProperties["stroke"];
};

export default function OneWayArrow(props: OneWayArrowProps) {
  return (
    <CustomEdge
      {...props}
      markerEnd={MarkerType.ArrowClosed}
      style={{
        strokeWidth: 2,
        stroke: props.stroke ?? "#FF0072",
      }}
    />
  );
}
```

I have inspected the element and I see that the `marker-end="arrowclosed"` already but something it doesn't show.
```
<g class="react-flow__edge react-flow__edge-onewayArrow nopan" tabindex="0" role="button" data-testid="rf__edge-reactflow__edge-1-2" aria-label="Edge from 1 to 2" aria-describedby="react-flow__edge-desc-1">
<path d="M372.03619909502265,-62.331825037707404 C372.03619909502265,40.24132730015083 479.7058823529411,40.24132730015083 479.7058823529411,142.81447963800906" fill="none" class="react-flow__edge-path" marker-end="arrowclosed" style="stroke-width: 2; stroke: rgb(255, 0, 114);"></path>
<path d="M372.03619909502265,-62.331825037707404 C372.03619909502265,40.24132730015083 479.7058823529411,40.24132730015083 479.7058823529411,142.81447963800906" fill="none" stroke-opacity="0" stroke-width="20" class="react-flow__edge-interaction"></path>
</g>
```

Version:
- reactflow@^11.10.1

**Comment by bcakmakoglu** [comment]
What you're trying doesn't work because the `markerEnd` *prop* is not the enum / obj definition of a marker.
You need to pass that with your edge definition.

```ts
{
	id: 'e1-2',
	type: 'custom',
	source: '1',
	target: '2',
	// pass the marker here!
	markerEnd: MarkerType.ArrowClosed,
}
```

After that you will receive a "url" for that marker as a prop for your custom edge

```tsx
function CustomEdge({ markerEnd }) {
	console.log(markerEnd) // logs something like `"url(#marker_arrow_closed)"`
}
```

Pass this string as `markerEnd` to your `path` elements.

**Reply by mike-dev-dsd** [reply]
markerEnd: { type: MarkerType.Arrow }  , If anyone got nothing TRY TO USE THIS instead.

**Comment by robcornelius** [comment]
I have spent several hours bashing my head against this. I couldn't find how to do this anywhere in the documentation. As its such a common use case perhaps someone could add it to the documentation

**Comment by anselanza** [comment]
I don't understand this one, either. I'm not even trying to make a "custom" edge. Just a standard one.

Edges defined like this:
```
const initialEdges: ReactFlowProps["edges"] = [
  {
    id: "n1-n2",
    source: "n1",
    target: "n2",
    markerEnd: "arrow",
    label: "diagnosed-high-intensity",
    animated: true,
  },
  {
    id: "n1-n3",
    source: "n1",
    target: "n3",
    label: "diagnosed-low-intensity",
    markerEnd: "arrow",
    animated: true,
  },
];
```

Very standard ReactFlow component like this:
```
<ReactFlow nodes={initialNodes} edges={initialEdges}>
        <Background />
        <Controls />
      </ReactFlow>
```

I should see arrows, right? But instead I see:
<img width="624" height="294" alt="Screenshot 2025-10-09 at 15 25 10" src="https://github.com/user-attachments/assets/ffdc472c-9813-4256-b1e5-dd22d40ce38e" />

**Reply by anselanza** [reply]
To answer my own question. This worked:

```
const arrowConnector: EdgeMarker = {
  type: MarkerType.ArrowClosed,
  width: 30,
  height: 30,
};

const initialEdges: ReactFlowProps["edges"] = [
  {
    id: "n1-n2",
    source: "n1",
    target: "n2",
    markerEnd: arrowConnector,
    label: "diagnosed-high-intensity",
    animated: true,
  },
  {
    id: "n1-n3",
    source: "n1",
    target: "n3",
    label: "diagnosed-low-intensity",
    markerEnd: arrowConnector,
    animated: true,
  },
];
```

So... I guess that adding the string `"arrowClosed"` is not enough, even though it passes TypeScript check for the type `EdgeMarkerType`. It's also odd to me that the default `width` and `height` appear to be `0`, which apparently is the reason that the arrow heads are invisible by default.
